File: factors.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  vector<int> factors;
  int number = 64;

  // O(sqrt(n))
  for (int i = 1; i * i <= number; i++) {
    if (number % i == 0) {
      factors.push_back(i);
      if (i != number / i) {
        factors.push_back(number / i);
      }
    }
  }

  // O(log(n))
  sort(factors.begin(), factors.end());

  // O(n)
  for (auto i : factors)
    cout << i << " ";
}

File: rust/factors.rs
fn main() {
    let mut factors: Vec<i32> = Vec::new();
    let number: i32 = 64;

    let limit = (number as f64).sqrt() as i32;

    for i in 1..=limit {
        if number % i == 0 {
            factors.push(i);
            if i != number / i {
                factors.push(number / i);
            }
        }
    }

    factors.sort();

    for i in factors {
        print!("{} ", i);
    }
}
